src/data/indexHostSingleton.ts

```diff
@@ -12,7 +13,7 @@ export const IndexHostSingleton = (function () {
 
   const _describeIndex = async (
     config: PineconeConfiguration,
-    indexName: string
+    indexName: IndexName
   ): Promise<string> => {
     if (!indexOperationsApi) {
       indexOperationsApi = indexOperationsBuilder(config);
```

Comment:
The `if (!host) { ... }` line below on ln 25 should be sufficient from preventing empty host values when the `_describeIndex` function is used internally.